tritium-registry/src/main/java/com/palantir/tritium/metrics/registry/TaggedMetricRegistry.java

```diff
@@ -60,16 +62,58 @@
 
     Histogram histogram(MetricName metricName, Supplier<Histogram> histogramSupplier);
 
+    /**
+     * Returns existing gauge metric for the specified metric name or empty if none has been registered.
+     *
+     * @implNote Implementations should override this method with a more efficient mechanism.
+     *
+     * @param metricName metric name
+     * @return gauge metric or empty if none exists for this name
+     */
+    @SuppressWarnings("unchecked")
+    default <T> Optional<Gauge<T>> gauge(MetricName metricName) {
+        Metric metric = getMetrics().get(metricName);
+        if (metric instanceof Gauge) {
+            return Optional.of((Gauge<T>) metric);
+        }
+        return Optional.empty();
+    }
+
     /**
      * Returns existing or new gauge metric for the specified metric name.
      *
      * @param metricName metric name
      * @param gauge gauge
      * @return gauge metric
+     *
+     * In most cases, one typically wants {@link #registerWithReplacement(MetricName, Gauge)} as gauges may
+     * retain references to large backing data structures (e.g. queue or cache), and if a gauge is being registered
+     * with the same name, one is likely replacing that previous data structure and the registry should not retain
+     * references to the previous version. If lookup of an existing gauge is desired, one should use
+     * {@link #gauge(MetricName)}.
      */
     // This differs from MetricRegistry and takes the Gauge directly rather than a Supplier<Gauge>
+    // @Deprecated
     <T> Gauge<T> gauge(MetricName metricName, Gauge<T> gauge);
```

Comment:
@carterkozak interested in your thoughts here -- I can't currently deprecate this method yet because of the cyclical dependency between `tritium`  & `metric-schema` as metric schema will generate code that uses this deprecated method (e.g. `com.palantir.tritium.metrics.jvm.OsMetrics#loadNorm1`) so the build will fail with deprecation warnings. We could release tritium without deprecating this method, then update metric-schema to have a minimum required tritium version that provides `registerWithReplacement` and use that, though we have to be careful both metric-schema & tritium versions are bumped in concert. Another option would be to move metric-schema into tritium to avoid the cyclical project dependency and version them together (slightly related to #474)

Reply:
I'm fine with staging the rollout & deprecation, mainly just don't want to cause unnecessary pain and breaks, especially going into the end of year.